### DISCUSSION TITLE
Maybe there is a little bug while simplifying list

### DISCUSSION THREAD

**Discussion by hairui** [scandum/tintin#94] [Bugs]
I happened to write a script like 

```
#var lstA {{1}{a}{2}{b}{3}{}{4}{}};

#var lstB {{1}{a}{2}{b}{3}{}{4}{}{5}{d}};

#list {lstA} {size} {szA1};
#list {lstB} {size} {szB1};
#showme $szA1 $szB1;

#list {lstA} {simplify} {};
#list {lstB} {simplify} {};

#list {lstA} {size} {szA2};
#list {lstB} {size} {szB2};
#showme $szA2 $szB2;

#showme {$lstA};
#showme {$lstB};

```

And the result is 

```

4 5
0 0
a;b;;
a;b;;;d


```
It seemed that while two "null" value present together at the end of the list, one or more null item would be lost while doing #list simplify
And this had caused some problem for my bot.

And another quesetion is how can I get the size of the list after simplify?

**Comment by scandum** [comment]
There are alternatives to simplify, like using:
```
#foreach {$lstA[]} {val} {#showme ($val)}
```

After you use simplify the only way to get the size would be using `#list lstA create {$lstA}`, though that doesn't work properly with your example, I'll look into that.

**Reply by hairui** [reply]
Yes ,there are many ways to avoid causing this little bug.
Just report it.
